File: app/src/main/java/com/fox/algorithm/数据结构与算法/leetcode/二叉树/_226翻转二叉树.java
package com.fox.algorithm.数据结构与算法.leetcode.二叉树;

/**
 * @Author Fox
 * @Date 2021/3/24 20:38
 */
/*https://leetcode-cn.com/problems/invert-binary-tree/
 */
class _226翻转二叉树 {

    public TreeNode invertTree(TreeNode node) {
        if(node == null) return null;

        TreeNode temp = node.left;
        node.left = node.right;
        node.right = temp;

        invertTree(node.left);
        invertTree(node.right);
        return node;
    }
}

File: app/src/main/java/com/fox/algorithm/数据结构与算法/leetcode/二叉树/226翻转二叉树.kt
package com.fox.algorithm.数据结构与算法.leetcode.二叉树

/**
 * @Author fox
 * @Date 2020/3/3 17:59
 */

fun invertTree(root: TreeNode?): TreeNode? {
    if (root == null) {
        return null
    }
    val temp = root.left
    root.left = root.right
    root.right = temp
    invertTree(root.left)
    invertTree(root.right)
    return root
}
